### PR TITLE
4413 - Debts datatable

### LINKED ISSUE #4413: Debts datatable update
**What we're after:**
We're making improvements to the debts datatable prior to launching. Following [data](https://github.com/fecgov/openFEC/issues/4758) and [design](https://github.com/fecgov/fec-cms/issues/4414) updates, will be ready for front-end to update the debts datatable. 

**Designs:** https://github.com/fecgov/fec-cms/issues/5658#issuecomment-1548025225

**Action items:**
 - [x] Data columns modified/added
 - [x] Additional data added in details panel
 - [x] Coverage date filter added to filter panel

**Completion criteria:**
- [x] Datatable matches the designs
- [x] [Followup ticket has been updated with notes for launch](https://github.com/fecgov/fec-cms/issues/5660)

### PR DESCRIPTION
**Pull request by rfultz**
## Summary

- Resolves #4413 

Adding the debts datatable and links, behind a feature flag.

### Required reviewers

- UX
- Front-end
- Back-end?
- SME (not on localhost)
- Content? (not on localhost)

## Impacted areas of the application

- /data/browse-data/
- /data/debts/
- committee profile pages (or anywhere a link to the debts tables will be)

## Screenshots

The browse-data page (Debts and "and debts" will only appear when the debts feature flag is on)
image

The debts tables (only available when the debts feature flag is on)
image

The bottom of a committee profile page. The debts-owed-by should only be a link if the feature flag is on—the numbers haven't been affected
image

## Related PRs

Related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()

## KNOWN ISSUES / QUESTIONS
- I need content for [the Debts section](http://127.0.0.1:8000/data/browse-data/?tab=loans-debts)
- Some candidate profiles don't have `committee_id` when building their links so I've made that link conditional in that `?committee_id=` only gets added if `committeed_id` exists

## How to test

- Pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Check the [Browse data](http://127.0.0.1:8000/data/browse-data/) page
- Check the [Loans and debts](http://127.0.0.1:8000/data/browse-data/?tab=loans-debts) tab/section of that page
- Click over to the [Debts datatable](http://127.0.0.1:8000/data/debts/)
- Play with the filters
- Click the data rows or their icon in the right column
- Click to a committee profile
- Check the profile's Financial summary section (,review the content generally?), check the debts/loans section at the bottom (value calculation hasn't changed, but the owed-by numbers should be a link now
- Click the owed-by link to make sure the datatable is filtering properly
- Extra credit: if you'd like to test the feature flag, edit `fec/fec/settings/base.py` and
   - edit line 90 (`FEATURES['debts']`) to be `False`, or
   - comment-out that line and line 72 (`'debts': `…)

### CODE REVIEW

**Comment by cnlucas** [comment]
Will debts be added under loans here or maybe the loans section can be loans and debts?

Screen Shot 2023-06-12 at 10 38 07 AM

**Comment by JonellaCulmer** [comment]
@rfultz Here's the intro text for the debts datatable on the browse data page. 

Search debts owed by committees. Search by committee, creditor/debtor's name, coverage date, beginning balance and ending balance.

cc: @dorothyyeager

File: fec/data/templates/macros/tables.jinja [hunk 1]
```diff
@@ -1,72 +1,85 @@
-{% macro summary(data, committee_id, cycle, office_or_committee, aggregate_cycles) %}
+{%- macro summary(data, committee_id, cycle, office_or_committee, aggregate_cycles) -%}
```
Comment (rfultz) [inline fec/data/templates/macros/tables.jinja:1]: All these new dashes are telling Jinja to delete the whitespace on that side of the code

**Review by johnnyporkchops — CHANGES_REQUESTED** [review]

File: fec/data/urls.py [hunk 2]
```diff
@@ -56,7 +56,6 @@
     )
 
 if settings.FEATURES.get('debts'):
```
[the remark this answers is not in the document]
Comment (rfultz) [reply fec/data/urls.py:58]: As far as I know. @JonellaCulmer ?

[the remark this answers is not in the document]
Comment (rfultz) [reply fec/data/urls.py:58]: To be clear, present / true shows the feature, which absent / negative switches it off

**Review by cnlucas — APPROVED** [review]
Works great, thank you

**Review by JonellaCulmer — APPROVED** [review]
This looked good from my perspective, thanks @rfultz

**Pull request merged**